### ISSUE TITLE
Inconsistent treatment of cookbooks_path in Chef Solo config

### ISSUE THREAD

**Issue by jimmycuadra** [hashicorp/vagrant#1570]
## Issue

Vagrant is currently enforcing the presence of the directories defined for the `cookbooks_path` key of a Chef Solo config, but only if it's been set explicitly in the Vagrantfile.
## Expected

Assuming, my Chef Solo cookbooks are stored in the directory "cookbooks," the following configuration should be a no-op:

``` ruby
config.vm.provision :chef_solo do |chef|
  chef.cookbooks_path = "cookbooks"
end
```
## Actual

The explicit declaration of a `cookbooks_path`, even if it's the same as what Vagrant will default to, causes the config validation to fail with the message:

```
chef solo provisioner:
* Cookbook path doesn't exist: /path/to/chef-solo/cookbooks
```

This is happening in `VagrantPlugins::Chef::Config::ChefSolo#validate`.
## Detail

At the very least, this behavior should be consistent, although enforcing the presence of the cookbooks directories seems of dubious value in general. When the Chef Solo provisioner runs, it simply skips sharing any directories which do not exist on the host (`VagrantPlugins::Chef::Provisioner::ChefSolo#expanded_folders`), which seems to negate any value of having enforced their existence during config validation.

Enforcing the existence of the cookbooks directory is also annoying when working with a plugin that manages your cookbooks, such as [vagrant-librarian-chef](https://github.com/jimmycuadra/vagrant-librarian-chef) (which is how this came up in the first place), where the cookbooks directory is intended to be added to the .gitignore file and managed through a "Cheffile" instead.
## tl;dr

Please consider removing the requirement that the Chef Solo cookbooks directory exists during config validation. If that's not possible, don't enforce it when the user explicitly sets it to the default value.

Thanks!
